src/web/mjs/connectors/Gntai.mjs

```diff
@@ -1,57 +1,51 @@
-import BloggerManga from './templates/BloggerManga.mjs';
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
 
-/**
- *
- */
-export default class Gntai extends BloggerManga {
+export default class Gntai extends Connector {
 
-    /**
-     *
-     */
     constructor() {
         super();
         super.id = 'gntai';
         super.label = 'GNTAI';
         this.tags = [ 'hentai', 'spanish' ];
-        this.url = 'http://www.gntai.xyz';
+        this.url = 'http://www.gntai.net';
+    }
+
+    async _getMangaFromURI(uri) {
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, 'main header.entry-header h1');
+        return new Manga(this, uri.pathname, data[0].textContent.trim());
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        let request = new Request(new URL(this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'div.navigation ul li:last-of-type a');
+        let pageCount = parseInt(data[0].text.trim());
+        for(let page = 1; page <= pageCount; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
 
-        this.path = '/page';
-        this.feed = 'default';
-        this.queryMangasPerPage = 50;
-        this.queryMangasPageCount = 'div#nav-wrapper select#select-nav-pag option:last-of-type';
+    async _getMangasFromPage(page) {
+        let request = new Request(new URL(`/page/${page}/`, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'main article div.chapter-thumb > a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.title.trim()
+            };
+        });
     }
 
-    /**
-     *
-     */
-    _getChapterList( manga, callback ) {
-        Promise.resolve()
-            .then( () => {
-                let chapterList = [ {
-                    id: manga.id,
-                    title: manga.title,
-                    language: ''
-                } ];
-                callback( null, chapterList );
-            } )
-            .catch( error => {
-                console.error( error, manga );
-                callback( error, undefined );
-            } );
+    async _getChapters(manga) {
+        return [ Object.assign({ language: '' }, manga) ];
```

Comment:
I don't get that part. Are you just returning the manga with a language flag ? How does the chapters get retrieved ?

Reply:
Hentai websites are usually one-shots without chapters, the manga is basically the one shot chapter.
It creates a new object which acts as chapter containing a `language` property and then assigned the `id` and `title` property from the manga to it.

Reply:
Samples for clarification of `Object.assign`:

```javascript
let source = { b: 99, x: 0 };
let target = { a: 1, b: 2 };
Object.assign(target, source);
console.log(source, target);
```

```javascript
let manga = { id: '/comic/my-little-pony', title: 'My Big Pony' };
let chapter = { language: 'pony' };
Object.assign(chapter, manga);
console.log(chapter);
```

Reply:
Object assign was ok, it was just the fact that the manga was returned. The previous statement clariify everything :)